### DISCUSSION TITLE
How to set this with a ddns like DuckDDNS?

### DISCUSSION THREAD

**Discussion by hollow-frenk** [nextcloud/all-in-one#612] [Questions] [answered]
Hi, I just installed ubuntu 22.04 from scratch, set everything related to the ddns but a doubt arises, that is, does this version of nextcloud work with ddns or only with real domains? If it works with ddns, is there anything to set up and/or permissions to give? Thank you for your possible reply and I hope for a detailed guide. ❤️

**Comment by szaimen** [comment, marked as answer]
Hi, yes it should work with duckdns. Basically you need to set up ddns by e.g. following https://gist.github.com/taichikuji/6f4183c0af1f4a29e345b60910666468, port-forward the important ports and set up AIO by following the instructions.

**Reply by hollow-frenk** [reply]
Thank you, the ddns is already set, only nextcloud is missing. Small question from a Linux novice, can you make sure that the backups created by the nextcloud aio system are automatic (for example every 24 hours) and that they are saved only on an external disk? If so, in what format should I format the external disk and how could you do the above thing step by step?

**Reply by szaimen** [reply]
> can you make sure that the backups created by the nextcloud aio system are automatic (for example every 24 hours)

Yesy there is an option for this in the backup and restore section.



> and that they are saved only on an external disk?

Yes. See 
![Screenshot_20220510-131024_Brave](https://user-images.githubusercontent.com/42591237/167615657-86b6e35a-d33c-4563-8c08-6ae4717f994a.png)

For how to format the disk just use your favorite search engine.

**Reply by hollow-frenk** [reply]
Thanks for the comprehensive answers, I installed everything but on the configuration screen I didn’t notice the option for automatic backups... Where is he? Then, logging in to the mobile app I realized I only had 76.61GB of cloud space even though I have a 1TB HDD available... What’s up? 👀

Edit: Oh, yes LVM 🤦🏻‍♂️

**Reply by szaimen** [reply]
> but on the configuration screen I didn’t notice the option for automatic backups... Where is he?

Did you enter the host location in the backup and restore section and create your first backup? Afterwards will the automatic backup be shown as option for you in that section



> Then, logging in to the mobile app I realized I only had 76.61GB of cloud space even though I have a 1TB HDD available... What’s up? 👀
> 
> Edit: Oh, yes LVM

You probably made a mistake somewhere...

**Reply by hollow-frenk** [reply]
Oh, okay, thanks ❤️

**Reply by hollow-frenk** [reply]
Ok, I just have two more questions.

1. Is it possible to change the username “admin”? If so, how?

2. Can I connect the OnlyOffice server provided by you to the desktop client? 🤔

P.s. As for the first question, if you are not possible, please add this feature. ❤️

**Reply by szaimen** [reply]
> 1. Is it possible to change the username “admin”? If so, how?

This is possible by simply creating a new admin in Nextcloud after the installation is done and deleting the `admin` user.

**Reply by szaimen** [reply]
> 2\. Can I connect the OnlyOffice server provided by you to the desktop client? 🤔

I dont understand

**Comment by nikita-fuchs** [comment]
Strangely, with 3 different ddns providers I get 3 different errors. I
1. ran the docker command from https://nextcloud.com/de/blog/how-to-install-the-nextcloud-all-in-one-on-linux/
2. forwarded the mentioned ports in my router and tested successfully with telnet
3. opened the https url 
 With duckdns, I get this error:


```
[Sun Jul 30 20:33:05.817938 2023] [core:notice] [pid 115:tid 548424269920] AH00094: Command line: 'httpd -D FOREGROUND'
{"level":"error","ts":1690749225.2060425,"logger":"http.acme_client","msg":"challenge failed","identifier":"<censored>.duckdns.org","challenge_type":"http-01","problem":{"type":"urn:ietf:params:acme:error:connection","title":"","detail":"2.203.144.253: Fetching http://<censored>.duckdns.org/.well-known/acme-challenge/P4ctuX2CJr_LgXaiCH1RlYzvFpPFnxVZL6NG_EY39f4: Error getting validation data","instance":"","subproblems":[]}}
{"level":"error","ts":1690749225.2068815,"logger":"http.acme_client","msg":"validating authorization","identifier":"<censored>.duckdns.org","problem":{"type":"urn:ietf:params:acme:error:connection","title":"","detail":"2.203.144.253: Fetching http://<censored>.duckdns.org/.well-known/acme-challenge/P4ctuX2CJr_LgXaiCH1RlYzvFpPFnxVZL6NG_EY39f4: Error getting validation data","instance":"","subproblems":[]},"order":"https://acme-v02.api.letsencrypt.org/acme/order/1232741916/198353093666","attempt":1,"max_attempts":3}
{"level":"error","ts":1690749225.2076628,"logger":"tls.obtain","msg":"could not get certificate from issuer","identifier":"<censored>.duckdns.org","issuer":"acme-v02.api.letsencrypt.org-directory","error":"HTTP 400 urn:ietf:params:acme:error:connection - 2.203.144.253: Fetching http://<censored>.duckdns.org/.well-known/acme-challenge/P4ctuX2CJr_LgXaiCH1RlYzvFpPFnxVZL6NG_EY39f4: Error getting validation data"}
{"level":"error","ts":1690749225.2077765,"logger":"tls.obtain","msg":"will retry","error":"[<censored>.duckdns.org] Obtain: [<censored>.duckdns.org] solving challenge: <censored>.duckdns.org: [<censored>.duckdns.org] authorization failed: HTTP 400 urn:ietf:params:acme:error:connection - 2.203.144.253: Fetching http://<censored>.duckdns.org/.well-known/acme-challenge/P4ctuX2CJr_LgXaiCH1RlYzvFpPFnxVZL6NG_EY39f4: Error getting validation data (ca=https://acme-v02.api.letsencrypt.org/directory)","attempt":1,"retrying_in":60,"elapsed":1.897703054,"max_duration":2592000}
```

@szaimen does the ddns have to have some 'minimum age' ? because it's brand new. or is there another reason for this error?

**Reply by szaimen** [reply]
Hi, see https://github.com/nextcloud/all-in-one/discussions/2105
